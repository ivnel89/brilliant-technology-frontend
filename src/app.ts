const greeting: String = "Bla"
console.log('Hello world!'+ greeting);

export const add = (a: number, b: number) => {
    return a + b;
}