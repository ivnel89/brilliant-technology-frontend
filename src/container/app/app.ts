import Api from "../../api";
import * as $ from "jquery";
import { Article } from "../../api/contract";

const api = new Api();

export const fetchArticlesAndDisplay = () => api.fetchAllArticles().then((articles: Array<Article>) => {
    const articleElements: Array<JQuery<HTMLElement>> = articles.map(
      (article) =>
        $(`
        <li>
            <a href="/article?id=${article.id}">
                ${article.content.slice(0, 20)}
            </article>
        </li>
        `)
    );
    $(`#root`).append($(`<ol></ol>`).append(articleElements));
});

fetchArticlesAndDisplay();