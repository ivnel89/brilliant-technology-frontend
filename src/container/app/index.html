<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8"/>
    <title>My Article Website</title>
    <link rel="stylesheet" href="../../styles.css" />
    <script type="module" src="./app.ts"></script>
  </head>
  <body>
    <div id="root"></div>
  </body>
</html>