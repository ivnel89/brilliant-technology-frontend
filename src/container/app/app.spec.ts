import { fetchArticlesAndDisplay } from './app';
import * as $ from "jquery";

jest.mock('../../api', () => {
    return {
        default: jest.fn().mockImplementation(() => {
            return {
              fetchAllArticles: jest.fn().mockResolvedValue([
                {
                  author: {
                    displayPicture: "www.example.com/img.png",
                    firstName: "Ben",
                    id: "mockAuthorId",
                    lastName: "Cook",
                  },
                  comments: [],
                  content: "Title",
                  createdDate: new Date(),
                  id: "mockArticleId",
                },
              ]),
            };   
        })
    }
  });
describe(`fetchArticlesAndDisplay`, () => {
    it(`should add article links to the DOM`, async () => {
        document.body.innerHTML = `
        <div id="root"></div>
      `;
        await fetchArticlesAndDisplay();
        expect($("#root").html()).toMatchSnapshot()
    } )
})