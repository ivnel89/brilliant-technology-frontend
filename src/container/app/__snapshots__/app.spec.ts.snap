// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchArticlesAndDisplay should add article links to the DOM 1`] = `
"<ol><li>
            <a href=\\"/article?id=mockArticleId\\">
                Title
            
        </a></li></ol>"
`;
